sdk/messaging/azservicebus/internal/errors.go

```diff
@@ -208,6 +198,59 @@ func shouldRecreateConnection(ctxForLogging context.Context, err error) bool {
 	return shouldRecreate
 }
 
+type recoveryKind string
+
+const RecoveryKindNone recoveryKind = ""
+const RecoveryKindFatal recoveryKind = "fatal"
+const RecoveryKindLink recoveryKind = "link"
+const RecoveryKindConn recoveryKind = "connection"
+
+type ServiceBusError struct {
+	inner        error
+	RecoveryKind recoveryKind
+}
+
+func (sbe *ServiceBusError) String() string {
+	return sbe.inner.Error()
+}
+
+func (sbe *ServiceBusError) AsError() error {
+	return sbe.inner
+}
+
+// ToSBE wraps the passed in 'err' with a proper error with one of either:
+// - `fatalServiceBusError` if no recovery is possible.
+// - `serviceBusError` if the error is recoverable. The `recoveryKind` field contains the
+//   type of recovery needed.
+func ToSBE(loggingCtx context.Context, err error) *ServiceBusError {
+	if err == nil {
+		return nil
+	}
+
+	sbe := &ServiceBusError{
+		inner:        err,
+		RecoveryKind: GetRecoveryKind(loggingCtx, err),
+	}
+
+	return sbe
+}
+
+func isPermanentNetError(err error) bool {
+	var netErr net.Error
+
+	if errors.As(err, &netErr) {
+		temp := netErr.Temporary()
```

Comment:
This is unreliable and should be avoided.  E.g. a DNS hiccup would return true here but it's really transient.

Reply:
In fact I'd yank this entire func.

Reply:
Removed. Basically, just doing what you noted in another spot - treat any net.Error as transient.

Reply:
I thought `isPermanentNetError` was going away?

Reply:
OH maybe it's going away in the second PR?

Reply:
(Yeah, I'm deleting it in the next PR).